app.js

```diff
@@ -1,13 +1,26 @@
 const express = require('express');
 const {getTopics} = require('./endpoints/controllers/topics.controller');
 const {getEndpoints} = require('./endpoints/controllers/api.controller');
+const {getArticleById} = require('./endpoints/controllers/articles.controller');
 const app = express();
 
 app.get('/api/topics', getTopics);
 app.get('/api', getEndpoints);
+app.get('/api/articles/:article_id', getArticleById);
 
 app.use((req, res, next) => {
     res.status(404).json({ message: 'Endpoint not found' });
   });
 
+  app.use((err, req, res, next) => {
+    if (err.status) {
+        res.status(err.status).send({ msg: err.msg });
+    } else if (err.code === '22P02') {
+        res.status(400).send({ msg: 'Invalid article ID' });
+    } else {
+        res.status(500).send({ msg: 'Internal Server Error' });
+    }
+});
```

Comment:
Nice, you can also split this up where you have a middleware handler for psql error codes, one for the custom errors and then a final one for the 500 internal server error as well